superset/views/core.py

```diff
@@ -3054,6 +3083,259 @@ def schemas_access_for_csv_upload(self):
+        """
+        db_path = os.getcwd() + "/" + db_name + ".db"
+        if os.path.isfile(db_path):
+            message = "Database file for {0} already exists, please choose a different name".format(
+                db_name
+            )
+            raise ValueError(message)
+        try:
+            item = SQLAInterface(models.Database).obj()
+            item.database_name = db_name
+            item.sqlalchemy_uri = "sqlite:///" + db_path
+            item.allow_csv_upload = True
+            # TODO check if SQL-injection is possible through add()
+            db.session.add(item)
+            db.session.commit()
+            return item
+        except Exception as e:
+            message = (
+                "Error when trying to create Database"
+                if isinstance(e, IntegrityError)
+                else str(e)
+            )
+            try:
+                if os.path.isfile(db_path):
+                    os.remove(db_path)
+                db.session.delete(item)
+                db.session.commit()
+            except OSError:
+                message = _(
+                    "Error when trying to create Database and Database-File could not be removed. "
+                    "Please contact your administrator to remove it manually"
+                )
+                pass
+            except Exception:
+                pass
+            stats_logger.incr("failed_csv_upload")
+            raise Exception(message)
+
+    def _get_existing_database(self, database_id: int, schema):
+        """Returns the database object for an existing database
+
+        Keyword arguments:
+        database_id -- the ID used to identify the database
+        schema -- the schema to be used
+
+        Raises:
+            ValueError: 1. If the schema is not allowed for csv upload
+                        2. If the database ID is not valid
+                        3. If there was a problem getting the schema
+        """
+        try:
+            database = self._get_database_by_id(database_id)
+            if not self._is_schema_allowed(database, schema):
+                message = _(
+                    "Database {0} Schema {1} is not allowed for csv uploads. "
+                    "Please contact your Superset administrator".format(
+                        database.database_name, schema
+                    )
+                )
+                raise ValueError(message)
+            return database
+        except Exception as e:
+            raise ValueError(e.args[0])
+
+    def _get_database_by_id(self, database_id: int) -> models.Database:
+        """ Returns the Database for the given ID
+
+        Keyword arguments:
+        database_id -- The ID which is used to identify the Database byo
+
+        Raises:
+            ValueError: If the database ID is not valid, i.e. matches no database
+        """
+
+        dbs = db.session.query(models.Database).filter_by(id=database_id).all()
```

Comment:
`id` is a primary key, so there is no risk of returning more than one row. This function can be removed and simplified to `db.session.query(models.Database).filter_by(id=database_id).one()`

Reply:
Should be fixed with [PR 33](https://github.com/Sascha-Gschwind/incubator-superset/pull/33). We changed it accordingly